### ISSUE TITLE
Mangeling and compressing in 4.3.0 is breaking code

### ISSUE THREAD

**Issue by ptrace** [terser/terser#449]
**Version (complete output of `terser -V` or specific git commit)**
```
terser 4.3.0
```
**Complete CLI command or `minify()` options used**
```
terser --compress --mangle -- t.js
```

**`terser` input**

```
function x() {
  try {
    let converted = '0'

    for (var i = 0; i < split_version.length; i ++) {
    }
    return converted
  } catch (err) {
    return ''
  }
}
```

In 4.2.1 the result was valid, in 4.3.0 the result is invalid, it outputs
```
function x(){try{let r="0";for(var r=0;r<split_version.length;r++);return r}catch(r){return""}}
```

Which beautifies to 
```
function x() {
    try {
        let r = "0";
        for (var r = 0; r < split_version.length; r++);
        return r
    } catch (r) {
        return ""
    }
}
```

Which is incorrect and triggers
```
Uncaught SyntaxError: Identifier 'r' has already been declared
```

On 4.2.1, it outputs correct code,

```
function x(){try{let t="0";for(var r=0;r<split_version.length;r++);return t}catch(r){return""}}
```

which beautifies to
```
function x() {
    try {
        let t = "0";
        for (var r = 0; r < split_version.length; r++);
        return t
    } catch (r) {
        return ""
    }
}
```

Which is OK.

**Comment by jacott** [comment]
I've tracked it down to commit c7cd7b754713380ddce7077ed9b6312522dac265 fix #335 by properly setting the catch-block's parent scope to be the parent scope of the try block, not the try block

**Comment by jacott** [comment]
This is my example which is breaking

```js
try {
  const {longname} = this;
  for(let i = 0; i < longname.length; ++i) {
    const item = longname[i];
  }
} finally {
}
```

which is converted to

```js
try{const{longname:n}=this;for(let n=0;n<longname.length;++n){longname[n]}}finally{}
```

**Comment by fabiosantoscode** [comment]
This has been fixed, internally it's the same bug as #452. Expect 4.3.1 in less than half an hour.
